pkg/prompb/types.pb.go

```diff
@@ -248,13 +248,25 @@ func (m *Sample) GetValue() float64 {
 	return 0
 }
 
+func (m Sample) GetVal() float64 {
+	return m.Value
+}
+
+func (m Sample) ExemplarLabels() []Label {
+	return nil
+}
```

Comment:
I didn't want to refer to the address of variables to just get the labels and value, and increase allocs and GC's work too much! Hence, having these functions saves some work. Will shift these to a separate file for easier maintenence.

Reply:
How does this increase allocs? It makes no sense.

Reply:
> Go compilers will allocate variables that are local to a function in that function’s stack frame. However, if the compiler 
cannot prove that the variable is not referenced after the function returns, then the compiler must allocate the variable on 
the garbage-collected heap to avoid dangling pointer errors. Also, if a local variable is very large, it might make more 
sense to store it on the heap rather than the stack.
If a variable has its address taken, that variable is a candidate for allocation on the heap. However, a basic escape 
analysis recognizes some cases when such variables will not live past the return from the function and can reside on the 
stack.

Hence, on a safer side, I avoided alloc.

https://medium.com/eureka-engineering/understanding-allocations-in-go-stack-heap-memory-9a2631b5035d

Reply:
I think there is confusion here. Addresses are simple fixed-size variables like ints that will be passed on the stack frame as arguments to the method. It will not go on the heap during calls.

Reply:
This does not seem like it would do anything

Reply:
I think a benchmark will be more definite here. Will try to cover/test during the upcoming automated benchmark section. For now, reverted.